### PR TITLE
378 add compathelper

### PR DESCRIPTION
**Pull request by cpaniaguam**
#378

### CODE REVIEW

File: .github/workflows/CompatHelper.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          version = "3"
+          Pkg.add(; name, uuid, version)
+        shell: julia --color=yes {0}
+      - name: "Run CompatHelper"
+        run: |
+          import CompatHelper
+          CompatHelper.main()
+        shell: julia --color=yes {0}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tdivoll) [inline .github/workflows/CompatHelper.yml:44]: Did you make new secrets or do we need to add the ones we already have in the repo?

File: .github/workflows/CompatHelper.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+jobs:
+  CompatHelper:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if Julia is already available in the PATH
+        id: julia_in_path
+        run: which julia
+        continue-on-error: true
+      - name: Install Julia, but only if it is not already available in the PATH
+        uses: julia-actions/setup-julia@v1
```
Comment (tdivoll) [inline .github/workflows/CompatHelper.yml:19]: update this to `setup-julia@v2`? Or let the dependabot flag it for updating?

**Review by tdivoll — CHANGES_REQUESTED** [review]

File: .github/workflows/CompatHelper.yml (hunk 2 above)
Comment (cpaniaguam) [reply .github/workflows/CompatHelper.yml:19]: I guess we can just change it here and now.

File: .github/workflows/CompatHelper.yml (hunk 1 above)
Comment (cpaniaguam) [reply .github/workflows/CompatHelper.yml:44]: I'll add a new one.

Comment (cpaniaguam) [reply .github/workflows/CompatHelper.yml:44]: No, it'll use the secret for the documentation build.

File: .github/workflows/CompatHelper.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if Julia is already available in the PATH
+        id: julia_in_path
+        run: which julia
+        continue-on-error: true
+      - name: Install Julia, but only if it is not already available in the PATH
+        uses: julia-actions/setup-julia@v2
+        with:
+          version: '1'
```
Comment (tdivoll) [inline .github/workflows/CompatHelper.yml:21]: Maybe this needs to be `version: '2'`? It seems redundant to declaring `v2` on line 19.

**Review by tdivoll — APPROVED** [review]

File: .github/workflows/CompatHelper.yml (hunk 3 above)
Comment (cpaniaguam) [reply .github/workflows/CompatHelper.yml:21]: That's actually the version of Julia that the action installs. See https://github.com/julia-actions/setup-julia

**Pull request merged**